vms/platformvm/txs/executor/camino_tx_executor.go

```diff
@@ -353,22 +354,66 @@ func (e *CaminoStandardTxExecutor) ExportTx(tx *txs.ExportTx) error {
 		return err
 	}
 
-	for _, output := range tx.ExportedOutputs {
+	multisigPositions := make(map[int]*multisig.Alias, len(tx.ExportedOutputs))
+	for i, output := range tx.ExportedOutputs {
 		out, ok := output.Out.(*secp256k1fx.TransferOutput)
 		if !ok {
 			return locked.ErrWrongOutType
 		}
 		for _, addr := range out.Addrs {
-			_, err := e.State.GetMultisigAlias(addr)
+			var err error
+			alias, err := e.State.GetMultisigAlias(addr)
 			if err != nil && err != database.ErrNotFound {
 				return err
 			} else if err == nil {
-				return errMsigAlias
+				// multisig alias found - sanity check: it has to be single address in Addrs
+				if len(out.Addrs) != 1 {
+					return errMsigNestedAliases
+				}
+				multisigPositions[i] = alias
 			}
 		}
 	}
 
-	return e.StandardTxExecutor.ExportTx(tx)
+	if err := e.StandardTxExecutor.ExportTx(tx); err != nil {
+		return err
+	}
+
+	for pos, alias := range multisigPositions {
+		putReqs := e.AtomicRequests[tx.DestinationChain].PutRequests
+		if pos >= len(putReqs) {
+			return errors.New("msig position out of range of put requests")
+		}
+		req := putReqs[pos]
+		var utxo avax.UTXO
+		_, err := txs.Codec.Unmarshal(req.Value, &utxo)
+		if err != nil {
+			return err
+		}
+
+		// sanity check
+		out := utxo.Out.(*secp256k1fx.TransferOutput)
```

Comment:
IMO we shouldn't fall back now to this simplified solution, the node itself supports complex msig definitions, mix from addresses and msigs. This means that we should be able to pass a bunch of aliases here, not only one to keep flexibility.
IMO the way to go is:
1.) collect all required msig aliases from outs (using new travers function)
2.) compare that these msig aliases are all passed in the tx.

Beside this I'm missing the check for tx.nonce == state.nonce, or how is the nonce working?
We had multiple approaches (ChainTime / Block / Nonce per alias (which then must be passed in TX) or a global nonce in state for all Aliases (incremented using internal logic).

Reply:
I saw TraverseOwners in camino_keychain but it seems to be switched off because of this check:
https://github.com/chain4travel/caminogo/blob/d0f69e2b223c48fd66af2dda37f493659c64aea5/vms/secp256k1fx/camino_fx.go#L114

This also leads to conclusion that we have no test proving the keychain code works correctly on combined / nested multisig. In that case I prefer to complete import and then focus to switch this functionality on.

Reply:
The `Nonce` on `multisig.Alias` is meant to change only when the underlying owners would change - which we don't have now (so at the moment Nonce == 0)